TeamSnapSDK/SDK/CollectionJSON/TSDKCollectionObject.h

```diff
@@ -77,6 +80,13 @@ NS_ASSUME_NONNULL_BEGIN
 - (void)setArray:(NSArray <NSString *> *_Nullable)value forKey:(NSString *)aKey;
 - (NSArray <NSString *> *_Nullable)getArrayForKey:(NSString *)key;
 
+- (id)collectionObjectForKey:(NSString *)key;
+- (void)removeCollectionObjectForKey:(NSString *)aKey;
+
+- (id _Nullable)changedValueForKey:(NSString *)aKey;
+- (void)setChangeValue:(id)value forKey:(NSString *)aKey;
```

Comment:
I think this should be should be `setChangedValueForKey` 
`changedValues` holds the original values when a value changes (so we can `undo` or get just the delta.

I also think the the case where we set a value to null should be `removeChangedValue:`

Reply:
Added `removeChangedValueForKey:`. In addition, allowed `setChangedValue:forKey:` to accept a nullable value that implicitly calls `removeChangedValueForKey:` for the best possible safety around that storage.